### ISSUE TITLE
Authentication Bypass

### ISSUE THREAD

**Issue by 649** [giekaton/php-metamask-user-login#28]
This bypass requires you to know the target wallet public key. The idea is simple, login and generate a valid JWT with a different wallet, and then update your own name once.

Capture your own request while updating the name, then simply change the wallet key to the target wallet and replay the request.
```
{"request":"updatePublicName","address":"<target wallet>","JWT":"<any valid token from any account>","publicName":"<target name>"}
```
This allows you to change anyone's name on the site. The problem is rooted in the fact that the JWT is decoupled from the user table that holds all the wallets and their nonces. One way to fix this is to include the JWT as a session key in the database and to use it in addition to the address in the UPDATE statement(s). 

This was tested on https://setinblock.com/php-metamask-user-login/ before creating this issue.

**Comment by vquelque** [comment]
That's correct. I think an easier fix would be to simply use the address contained in the JWT i.e., the address contained in the payload returned by the call to`jwt::Decode` instead of the address contained in the request payload.

**Comment by fulldecent** [comment]
Here is more discussion on Web3 login https://blog.phor.net/break-into-dapp-admin-accounts and a review of several large projects and how they do it
